### PR TITLE
Generate Badges for Manual Data with pre-selected images

### PR DESCRIPTION
**Pull request by gabru-md**
Changed an elif to if so that if a predefined image is used the the CSV is also checked for.
The error was caused because of the fact that elif was not executed if the if of image was encountered.
I've managed it and fixed the bug. There were several other bugs as well that I've fixed so as to implement proper functionality.
All the changes can be found in the commit section on thie PR

Fixes : https://github.com/fossasia/badgeyay/issues/314

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] I have added necessary documentation (if appropriate)

#### Changes proposed in this pull request:

Few changes but critical ones.

### CODE REVIEW

**Comment by gabru-md** [comment]
@djmgit @ParthS007 @vaibhavsingh97 @S2606 @mariobehling @jajodiaraghav  Please review.
And sorry for inconvenience

File: app/main.py [hunk 1]
```diff
@@ -3,14 +3,15 @@
...
-import uuid
+# import uuid
 import traceback
 import generate_csv_eventyay
 from svg_to_png import do_svg2png
 
 APP_ROOT = os.path.dirname(os.path.abspath(__file__))
 UPLOAD_FOLDER = os.path.join(APP_ROOT, 'static/uploads')
 BADGES_FOLDER = os.path.join(APP_ROOT, 'static/badges')
+SCRIPT = os.path.join(APP_ROOT, 'merge_badges.sh')
```
Comment (S2606) [inline app/main.py]: @gabru-md All the files were converted to python.So i dont think we need this line

Comment (gabru-md) [reply app/main.py]: okay!

**Comment by gabru-md** [comment]
@S2606 done.

**Comment by djmgit** [comment]
badgeyay

@gabru-md getting the above error when I am trying to upload a csv file and selecting a background from dorpdown.

**Comment by gabru-md** [comment]
@djmgit I've made the changes and now it works fine. Please let me know if any error persists as I guess that this is a tough bug to resolve due to inconsistency in `main.py` file. Therefore any new comments are welcomed.

**Comment by gabru-md** [comment]
Once I'm done with this just let me know I'll squash all my commits . Till then I'll be making individual commits.

**Comment by gabru-md** [comment]
@djmgit @S2606 @vaibhavsingh97 @ParthS007 Please Review if all the cases are working!

File: app/main.py [hunk 2]
```diff
@@ -66,67 +66,72 @@ def upload():
...
-            flash('Please select a CSV file to Upload!', 'error')
-            return redirect(url_for('index'))
 
     # If default background is selected
     if img != '':
         if(img == 'user_defined.png'):
             bg_color = request.form['bg_color']
             text_on_image = request.form['text_on_image']
             do_svg2png(img, 1, bg_color, text_on_image)
+            filename = img + '.csv'
```
Comment (jajodiaraghav) [inline app/main.py]: What is the point of adding this line here? The `filename = img + '.csv'` is going to execute in the next line anyway

Comment (gabru-md) [reply app/main.py]: sorry i did not actually see that one. will remove it

**Comment by gabru-md** [comment]
@S2606 @vaibhavsingh97 @jajodiaraghav Please Review 
This is a major bug fix so I'm looking for more reviews after it this can be merged.

**Comment by jajodiaraghav** [comment]
Approved! We should still look for a fix for Travis

**Comment by gabru-md** [comment]
@vaibhavsingh97 It was difficult for me to figure out such an important bug and rectify it. Can we please leave the travis fixing issue in another PR ?
Maybe someone else would like to figure it out.
Also I ambusy with my college's viva and all so I'll not be able to travis as of now.
I'll be squashing my commits no more than tomorrow morning. Then this must be merged.

**Comment by S2606** [comment]
@gabru-md Should I merge this?

**Comment by gabru-md** [comment]
@S2606 yes please.

**Pull request merged**